code/Control/UserDefinedFormController.php

```diff
@@ -421,7 +422,8 @@ public function process($data, $form)
                 // Include any parsed merge field references from the CMS editor - this is already escaped
                 // This string substitution works for both HTML and plain text emails.
                 // $recipient->getEmailBodyContent() will retrieve the relevant version of the email
-                $emailData['Body'] = SSViewer::execute_string($recipient->getEmailBodyContent(), $mergeFields);
+                $engine = Injector::inst()->create(TemplateEngine::class);
+                $emailData['Body'] = $engine->renderString($recipient->getEmailBodyContent(), ViewLayerData::create($mergeFields));
```

Comment:
The "EmailBodyContent" is a string that represents a template. It's defined by content authors. By default it should be ss template syntax and the ss template renderer will be used - but it could be twig or whatever-the-heck else so I've decided to fetch whatever's registered under `TemplateEngine` instead of explicitly instantiating `SSTemplateEngine` here.